provider/oci/environ.go

```diff
@@ -498,20 +498,25 @@ func (e *Environ) startInstance(
 		return nil, errors.Trace(err)
 	}
 
-	types := imgCache.SupportedShapes(args.InstanceConfig.Base)
-
 	defaultType := VirtualMachine.String()
 	if args.Constraints.VirtType == nil {
 		args.Constraints.VirtType = &defaultType
 	}
 
+	// Since the images for ARM machine will include "aarch64" on their
+	// name, we need to make sure we select the correct base that was
+	// taken from the parsed image name.
+	if arch == corearch.ARM64 {
+		args.InstanceConfig.Base.OS = UbuntuARMBase
+	}
+
```

Comment:
I have concerns about overloading the defined juju base.OS values with architecture details and the values ending up in state. Nor am I clear how this interacts with the architecture constraints.

What about updating imgCache.SupportedShapes to take an architecture value as well. The actual imgCache is local to the Oracle provider.

Reply:
indeed, this was a very bad idea. I changed the implementation, now we pass the architecture as parameter to the image functions. Also, we keep track of the image's architecture on the cache so we can select them using the passed arch parameter.